library/ssl_misc.h

```diff
@@ -1307,8 +1307,10 @@ void mbedtls_ssl_add_hs_msg_to_checksum( mbedtls_ssl_context *ssl,
                                          size_t msg_len );
 
 #if defined(MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED)
+#if !defined(MBEDTLS_USE_PSA_CRYPTO)
```

Comment:
Thanks for this change! I was not super happy with the state for this function before - mostly not called for PSA, except for DHE-PSK (and then with an early exit). I find it much clearer after this commit, thanks for the simplification!